src/Spice86.Core/Emulator/Devices/ExternalInput/Pic.cs

```diff
@@ -21,176 +23,288 @@
 /// <li>https://k.lse.epita.fr/internals/8259a_controller.html</li>
 /// </ul>
 /// </summary>
-public class Pic : DefaultIOPortHandler {
+public class Pic {
     private static readonly ILogger _logger = new Serilogger().Logger.ForContext<Pic>();
 
-    private const int MasterPortA = 0x20;
-
-    private const int MasterPortB = 0x21;
-
-    private const int SlavePortA = 0xA0;
+    private Machine _machine;
 
-    private const int SlavePortB = 0xA1;
+    private bool _initialized;
 
-    private static readonly Dictionary<int, int> _vectorToIrq = new();
+    private byte _baseInterruptVector;
 
-    private int _commandsToProcess = 2;
+    private int _initializationCommandsExpected = 2;
 
-    private int _currentCommand = 0;
+    private int _currentInitializationCommand = 0;
 
-    private bool _initialized = false;
+    private bool _master;
 
-    private byte _interruptMask = 0;
+    // 1 indicates the channel is masked (inhibited), 0 indicates the channel is enabled.
+    private byte _interruptMaskRegister = 0;
+    private byte _interruptRequestRegister = 0;
+    private byte _inServiceRegister = 0;
+    private byte _currentIrq = 0;
+    private byte _lowestPriorityIrq = 7;
+    private bool _specialMask = false;
+    private bool _polled = false;
+    private bool _interruptOngoing = false;
+    private SelectedReadRegister _selectedReadRegister = SelectedReadRegister.InterruptRequestRegister;
 
-    static Pic() {
-        // timer
-        _vectorToIrq.Add(8, 0);
-        // keyboard
-        _vectorToIrq.Add(9, 1);
-    }
-
-    public Pic(Machine machine, bool initialized, Configuration configuration) : base(machine, configuration) {
-        _initialized = initialized;
+    public Pic(Machine machine, bool master) {
+        _master = master;
+        _machine = machine;
     }
 
     /// <summary>
```

Comment:
obsolete documentation. It applied to the old code. Probably needs to be deleted.